script.js

```diff
@@ -1,4 +1,117 @@
+// global variables
+var gameMode = ""; //gameMode controls what type of game we play
+var gameStep = ""; //gameStep controls which stage of code to run
+var player1Dice = [];
+var player2Dice = [];
+var player1CurrentRoll = 0;
+var player2CurrentRoll = 0;
+var player1TotalScore = 0;
+var player2TotalScore = 0;
+var currentWinner = "";
+
+// create dice roll function
+var DiceRoll = function () {
+  diceNumber = Math.floor(Math.random() * 6) + 1;
+  return diceNumber;
+};
+
+// Creating side functions for dice rolling
+var TwoPlayerDiceRoll = function () {
+  var playerDice = [];
+  playerDice.push(DiceRoll());
+  playerDice.push(DiceRoll());
+  console.log(playerDice);
+  return playerDice;
+};
+
+//Score counter takes the dice numbers rolled and counts the total score depending on user input
+var ScoreCounter = function (input, totalScore, diceRoll) {
+  if (gameMode == "Two player" || gameMode == "Lowest combined number") {
+    if (input == "Dice 1") {
+      var currentRoll = 10 * diceRoll[0] + diceRoll[1];
+    } else if (input == "Dice 2") {
+      var currentRoll = 10 * diceRoll[1] + diceRoll[0];
+    }
+  }
+  if (gameMode == "Auto generate") {
+    if (diceRoll[0] >= diceRoll[1]) {
+      var currentRoll = 10 * diceRoll[0] + diceRoll[1];
```

Comment:
variables only need to be declared once, maybe put `var currentRoll;` after line 28?